jenkins/open-mpi-build-script.sh

```diff
@@ -179,6 +179,14 @@ fi
 # ./configure --help.  In prep for 5.0, we added a developer
 # requirement for pandoc to build (not required for dist tarballs),
 # with an explicit option to disable.
+# JMS: The requirement will shortly change from pandoc to Sphinx.
+# Keep pandoc for now, but also add sphinx so that our CI can test the
+# Sphinx PR before merging the removal of pandoc/addition of Sphinx.
+if test -f docs/requirements.txt; then
```

Comment:
This script runs every time a job starts.  So you're building a new environment 1) in the OMPI tree and 2) every build.  That's pretty wasteful.  I can't remember if we're making Spinx required (as in, must have, or can't build OMPI) or optional (like pandoc).

If it's optional like pandoc, I think following the pandoc "if the executable isn't already available, add the --disable flag".

If it's not optional, then we should do caching, similar to the autogen tools in the dist script.

Reply:
Jeff and I talked offline about this and decided that the right answer was just to install Sphinx on the build AMIs.  I did this today, so we shouldn't need this part of the commit (but may need other parts, below).